go/slice/slice_test.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2024 The Vitess Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package slice
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAll(t *testing.T) {
+	testCases := []struct {
+		name     string
+		input    []int
+		fn       func(int) bool
+		expected bool
+	}{
+		{
+			name:     "EmptySlice",
+			input:    []int{},
+			fn:       func(i int) bool { return i > 0 },
+			expected: true,
+		},
+		{
+			name:     "AllElementsTrue",
+			input:    []int{1, 2, 3},
+			fn:       func(i int) bool { return i > 0 },
+			expected: true,
+		},
+		{
+			name:     "SomeElementsFalse",
+			input:    []int{1, 2, 0},
+			fn:       func(i int) bool { return i > 0 },
+			expected: false,
+		},
+		{
+			name:     "SingleElementFalse",
+			input:    []int{0},
+			fn:       func(i int) bool { return i > 0 },
+			expected: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			result := All(tc.input, tc.fn)
+			assert.Equal(t, tc.expected, result, "All should return the expected result")
+		})
+	}
+}
+
+func TestAny(t *testing.T) {
+	testCases := []struct {
+		name     string
+		input    []int
+		fn       func(int) bool
+		expected bool
+	}{
+		{
+			name:     "ReturnsTrue",
+			input:    []int{1, 2, 3, 4, 5},
+			fn:       func(n int) bool { return n == 3 },
+			expected: true,
+		},
+		{
+			name:     "EmptySliceInput",
+			input:    []int{},
+			fn:       func(n int) bool { return n > 0 },
+			expected: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			result := Any(tc.input, tc.fn)
+			assert.Equal(t, tc.expected, result, "Any should return the expected result")
```

Comment:
I don’t think the 4th argument here provides much value. The same in the other places. Let’s remove that to further simplify.

Reply:
Okay! I I've removed the 4th argument. Please check.